docs/src/assets/scripts.js

```diff
@@ -188,7 +188,21 @@ const getWebpage = async (folder, hash, shallUpdateHistory) => {
     currentPage.bookPath = undefined;
     searchInput.focus();
     const searchResultsList = doc.getElementById('search-results-list');
-    searchInput.addEventListener('keyup', async (evt) => {
+    /*
+    https://www.freecodecamp.org/news/javascript-debounce-example/
+    */
+    function debounce(fn, timeoutInMilliseconds = 250) {
+      let timer;
+      return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(
+          () => { fn.apply(this, args); },
+          timeoutInMilliseconds
+        );
+      };
+    }
+    const processKeyUp = debounce(() => search());
```

Comment:
This is a huge nitpick... debounce is generally useful, so you should define it at the top-level, not inside of `getWebpage` and you should define `processKeyUp` just before using it, or not at all, and just put `debounce(search)` directly into the `addEventListener` call

Reply:
That makes sense!

Reply:
[843d7a5](https://github.com/reach-sh/reach-lang/pull/1224/commits/843d7a5980433a887df42e754c3ee044e3a0610c) defines `debounce` at the top-level.

Reply:
[cf47db5](https://github.com/reach-sh/reach-lang/pull/1224/commits/cf47db533f297118ee84961dfc8586222c95e765) removes `processKeyUp` altogether.